pkg/apis/core/validation/validation.go

```diff
@@ -4393,6 +4393,28 @@ func validatePodResourceConsistency(spec *core.PodSpec, fldPath *field.Path) fie
 		}
 	}
 
+	// Pod level hugepage limits must be always equal or greater than the aggregated
+	// container level hugepage limits, this is due to the hugepage resources being
+	// treated as a non overcommitable resource (request and limit must be equal)
+	// for the current container level hugepage behavior.
+	// This is also why hugepages overcommitment is not allowed in pod level resources,
+	// the pod cgroup values must reflect the request/limit set at pod level, and the
+	// container level cgroup values must be within that limit.
+	aggrContainerLims := resourcehelper.AggregateContainerLimits(&v1.Pod{Spec: *v1PodSpec}, resourcehelper.PodResourcesOptions{})
+	for resourceName, ctrLims := range aggrContainerLims {
+		if !helper.IsHugePageResourceName(core.ResourceName(resourceName)) {
+			continue
+		}
+
+		key := resourceName.String()
+		podSpecLimits := spec.Resources.Limits[core.ResourceName(key)]
```

Comment:
maybe short-circuit if the pod has no limit for this? we only sometimes default the pod-level limits so there may be no limit

```go
podSpecLimits, hasLimit := spec.Resources.Limits[core.ResourceName(key)]
if !hasLimit {
  continue
}
```

Reply:
Done, thank you

Reply:
Done, thank you